src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java

```diff
@@ -131,6 +177,57 @@ public void initialize() {
      */
     @Override
     public void dispose() {
+        disposeReceiveThread();
+        disposeSerialConnection();
+
+        if (watchdog != null && !watchdog.isCancelled()) {
+            watchdog.cancel(true);
+        }
+        logger.debug("Stopped ZWave serial handler");
+
+        super.dispose();
+    }
+
+    /**
+     * Stops resources when serial port is disconnected.
+     */
+    private void onSerialPortError(String errorMessage) {
+        logger.warn("Serial port error: '{}'", portId);
+        stopNetwork();
+        disposeReceiveThread();
+        disposeSerialConnection();
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                errorMessage + " [\"" + portId + "\"]");
+    }
+
+    /**
+     * Closes serial port connection
+     */
+    private void disposeSerialConnection() {
+        logger.trace("Disposing serial connection");
+        if (inputStream != null) {
+            IOUtils.closeQuietly(inputStream);
+            inputStream = null;
+        }
+        if (outputStream != null) {
+            IOUtils.closeQuietly(outputStream);
+            outputStream = null;
+        }
+
+        if (serialPort != null) {
+            serialPort.removeEventListener();
+            serialPort.close();
+            serialPort = null;
+        }
+        logger.trace("Serial connection disposed");
+    }
+
+    /**
+     * Closes receive thread
+     */
+    private void disposeReceiveThread() {
+        logger.trace("Disposing receive thread");
```

Comment:
Please make some of these messages debug level. I've not tried to run this, but some of these messages about closing down the serial connection should be raised so I can see what is happening. Currently they all seem to be trace.